### PR TITLE
Update unstructured document loaders

### PR DESCRIPTION
**Pull request by QuinnGT**
**Description:**

- Updated unstructured file loader with latest API changes
- Update unstructured folder loader with latest API changes
- Updated S3File loader with latest unstructured API changes
- Updated S3File loader to use new region lookup method
- Update S3File loader to use updated credential function
- Added support for `ocrLanguages` but should be updated to be just `languages` once langchain unstructured gets an updated

**Note:** 
Unstructured folder loader currently doesn't work with different file types like .png and .heic due to problems with their API. This is in progress for future releases.

Also, I would like to add S3Folder loader but I'll do that in a separate PR since langchainjs doesn't support it yet and I'll have to go direct to the sdk.

**Issues this addresses:**

- #1409
- #780

### CODE REVIEW

File: packages/components/nodes/documentloaders/Unstructured/UnstructuredFile.ts [hunk 1]
```diff
@@ -118,44 +281,57 @@ class UnstructuredFile_DocumentLoaders implements INode {
     async init(nodeData: INodeData, _: string, options: ICommonObject): Promise<any> {
         const filePath = nodeData.inputs?.filePath as string
         const unstructuredAPIUrl = nodeData.inputs?.unstructuredAPIUrl as string
-        const elementType = nodeData.inputs?.elementType as string
+        const strategy = nodeData.inputs?.strategy as UnstructuredLoaderStrategy
+        const encoding = nodeData.inputs?.encoding as string
+        const coordinates = nodeData.inputs?.coordinates as boolean
+        const xmlKeepTags = nodeData.inputs?.xmlKeepTags as boolean
+        const skipInferTableTypes = nodeData.inputs?.skipInferTableTypes as SkipInferTableTypes[]
```
Comment (HenryHengZJ) [inline packages/components/nodes/documentloaders/Unstructured/UnstructuredFile.ts:288]: for `multiOptions`, should we do `JSON.parse(skipInferTableTypes)` ?

**Comment by HenryHengZJ** [comment]
Thanks @QuinnGT this is so good!

There's a few other parameters we can add in future:
- multiPageSections
- combineUnderNChars
- newAfterNChars
- maxCharacters

File: packages/components/nodes/documentloaders/Unstructured/UnstructuredFile.ts (hunk 1 above)
Comment (QuinnGT) [reply packages/components/nodes/documentloaders/Unstructured/UnstructuredFile.ts:288]: Oh good catch. I made the change and tested.
Before: `"skip_infer_table_types": ["\"[\\\"pdf\\\",\\\"jpg\\\",\\\"pptx\\\"]\""],`
After: `"skip_infer_table_types": ["pdf", "jpg", "pptx"],`

**Comment by QuinnGT** [comment]
Thank you! I completely agree. Looking forward to the langchain upgrade so we can bring those additional params in.

**Comment by HenryHengZJ** [comment]
> Thank you! I completely agree. Looking forward to the langchain upgrade so we can bring those additional params in.

merged!

**Review by HenryHengZJ — APPROVED** [review]
looks good! are you still planning to add more changes or is it good to merge @QuinnGT ?

**Comment by QuinnGT** [comment]
Busy today but plan on adding the rest of those params tonight (utc-7).

**Comment by QuinnGT** [comment]
Hey @HenryHengZJ take a look at the latest changes. Just tested on multiple docs, fixed some errors, and got it stable.

**Comment by HenryHengZJ** [comment]
> Hey @HenryHengZJ take a look at the latest changes. Just tested on multiple docs, fixed some errors, and got it stable.

awesome thank you so much!

**Pull request merged**